SD_Card.h

```diff
@@ -0,0 +1,85 @@
+#include <SD.h>
+#include <SPI.h>
+
+#include "gps.h"
+#include "imu.h"
+
+#pragma once
+
+class SDCard {
+  private:
+    File m_sdCardDataFile;
+    static const int m_chipSelect = 4; // this is not the final value - SDCARD_SS_PIN is the correct one for the MKR Zero
+    static const String m_fileName = "rocketry_data";
+  public:
+    void writeToCSV( // this is equivalent 
+      const GPS::Coordinates& coords,
+      const IMU::vector3& accel,
+      const IMU::vector3& gyro,
+      float altitude,
+      int analogReading, // this is VOC
+      float humidity,
+      float temperature) {
+        // checks if SD card is open and good to be written to 
+        if (m_sdCardDataFile) {
```

Comment:
Oh, good idea, I didn't even think of that. Do we want to try `open` again if not?